tests/GhcSession.hs

```diff
@@ -113,11 +132,15 @@ compileModule ep opts = do
 
     handleSourceError (\e -> GHC.printException e >> return False) $ do
 
+    let target = case nb of
+          ProduceBuildOutput -> HscNothing -- AZ: what should this be?
```

Comment:
You know, I saw your conversation with ezyang and I didn't realize backpack packages with holes won't work with GHCi/HscInterpreted when we started this.

The way I see it we have two choices here either we tell ghc-mod which hscTarget it has to use and have it operate in a degraded capacity or we tell it to ignore such backpack packages altogether the latter would mean users couldn't even syntax check modules from such components.

The former would need investigating which target to use whereas the latter is more or less zero effort. I guess we can deduce that we need to change target/ignore a component based on `needsBuildOutput` for now so we can defer the decision until we get to integrating these changes into ghc-mod.